Assets/MixedRealityToolkit/MSBuild/Scripts/TemplateFiles.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.MSBuild
+{
+    /// <summary>
+    /// A helper class to manage (and locate) all the templates.
+    /// </summary>
+    public class TemplateFiles
+    {
+        private const string TemplateFilesFolderName = "MSBuildTemplates";
+        private const string MSBuildSolutionTemplateName = "SolutionTemplate.sln";
+        private const string SDKProjectFileTemplateName = "SDKProjectTemplate.csproj";
+        private const string PlatformPropsTemplateName = "Platform.Configuration.Template.props";
+        private const string EditorPropsTemplateName = "Editor.InEditor.Template.props";
+        private const string SpecifcPlatformPropsTemplateRegex = @"[a-zA-Z]+\.[a-zA-Z]+\.Template\.props";
+
+        private static TemplateFiles instance;
+
+        /// <summary>
+        /// Gets the singleton instance (created on demand) of this class.
+        /// </summary>
+        public static TemplateFiles Instance => instance ?? (instance = new TemplateFiles());
+
+        private string generatedCSMetaTemplateText;
+
```

Comment:
I think dead variable (would probably be worth just going through and cleaning whatever build warnings exist, as they should also end up being build errors in CI)